src/datasource/tracking.ts

```diff
@@ -0,0 +1,65 @@
+import { DataQueryRequest } from '@grafana/data';
+import { ZabbixMetricsQuery } from './types';
+import { reportInteraction } from '@grafana/runtime';
+
+export const trackRequest = (request: DataQueryRequest<ZabbixMetricsQuery>): void => {
```

Comment:
phew, this is a lot of code here 😄 .

i understand the reason.

still, i'd recommend one of two approaches:
1. only track some minimal info, like only the query-type
2. refactor this code:
    - here we can have a function that just builds the "report-interaction-object" from the request-object
       - we can write a bunch of unit-tests for it to be sure it's not crashing
    - and then call it from `datasource.ts` with something like `reportInteraction('something_something', extractProps(request))`


random note: when the zabbix code handles the `.queryType`, it seems to use these values to compare with: https://github.com/grafana/grafana-zabbix/blob/f7d84d1b1ff966c386627fa666cf884d5465ac71/src/datasource/constants.ts#L10-L16 (for example see here: https://github.com/grafana/grafana-zabbix/blob/f7d84d1b1ff966c386627fa666cf884d5465ac71/src/datasource/components/QueryEditor.tsx#L118 ), maybe we could use those values too?

Reply:
i think we should just go with option 1 for now then and if we decide in future we would like more info we could implement 2.